### PR TITLE
Document data model as a proto schema

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: assets/proto/element.proto [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  // The vertical/horizontal flip.
+  // Default: "no flip".
+  optional Flip flip = 10;
+
+  // The clipping mask of the element.
+  // Default: "no mask" or a simple rectangle mask.
+  optional Mask mask = 11;
+
+  // TODO: not sure if an extension or just optional fields are better here.
+  extensions 1000 to max;
```
Comment (wassgha) [inline assets/proto/element.proto:41]: I like extensions

**Review by wassgha — APPROVED** [review]
Agreed, let's figure out a way to keep these in sync

File: assets/proto/element.proto (hunk 1 above)
Comment (dvoytenko) [reply assets/proto/element.proto:41]: Am I using them correctly here? E.g. would I be able to do `elements = [new MediaElement(), new TextElement()]`, etc on the `repeated Element elements`? Also, it bugs me:

1. I can't have an extension w/o fields. E.g. I also want to have `ImageElement extends MediaElement`  that, for now, might not have any fields.
2. There are two things responsible for type: the field "type" and the type-base message, e.g. `TextElement`. Is there a better way to tie them together?

File: assets/proto/types.proto [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  CENTER = 4;
+  JUSTIFY = 5;
+}
+
+enum TextDecoration {
+  UNKNOWN_TEXT_DECORATION = 0;
+}
+
+enum BackgroundTextMode {
+  UNKNOWN_BACKGROUND_TEXT_MODE = 0;
```
Comment (barklund) [inline assets/proto/types.proto:47]: `NONE` here as an explicit setting?

File: assets/proto/types.proto [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+// Pattern with type = LINEAR|RADIAL|CONICAL.
+message GradientPattern {
+  message Stop {
+    required float position = 1;
+    required Rgba color = 2;
+  }
+  extend Pattern {
+    optional float alpha = 201 [default = 1];
+    repeated Stop stop = 202;
```
Comment (barklund) [inline assets/proto/types.proto:124]: Missing:
* `float rotation` for `LINEAR` and `CONIC`
* `Point center` for `RADIAL` and `CONIC`
* `Dimension size` for `RADIAL`

Comment (barklund) [reply assets/proto/types.proto:124]: (with `Point` being an `{ float x, float y }` struct and `Dimension` being an `{ float w, float h }` struct)

**Pull request closed without merging**